src/java.base/macosx/classes/jdk/internal/loader/ClassLoaderHelper.java

```diff
@@ -27,20 +27,16 @@
 
 import java.io.File;
 import java.util.ArrayList;
-import sun.security.action.GetPropertyAction;
+
+import jdk.internal.util.OperatingSystem;
+import jdk.internal.util.Version;
 
 class ClassLoaderHelper {
     private static final boolean hasDynamicLoaderCache;
     static {
-        String osVersion = GetPropertyAction.privilegedGetProperty("os.version");
-        // dynamic linker cache support on os.version >= 11.x
-        int major = 11;
-        int i = osVersion.indexOf('.');
-        try {
-            major = Integer.parseInt(i < 0 ? osVersion : osVersion.substring(0, i));
-        } catch (NumberFormatException e) {}
         // SDK 10.15 and earlier always reports 10.16 instead of 11.x.x
-        hasDynamicLoaderCache = major >= 11 || osVersion.equals("10.16");
+        hasDynamicLoaderCache = OperatingSystem.version()
+                .compareTo(new Version(10, 16)) >= 0;
```

Comment:
I wonder if this check can be removed too as 10.11 is the oldest macOS release that seems to be getting updates.

Reply:
10.15 supported as of JDK 20; will keep the check until that changes.

Reply:
This assignment could probably be moved into a static field initializer, as it’s now a simple expression:
```java
    // SDK 10.15 and earlier always reports 10.16 instead of 11.x.x
    private static final boolean hasDynamicLoaderCache = OperatingSystem.version()
            .compareTo(new Version(10, 16)) >= 0;
```